### PR TITLE
Add NDK references to source context setup

### PR DESCRIPTION
**Pull request by stefanosiano**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes
Fixes https://github.com/getsentry/sentry-docs/issues/6677

Added NDK references to source context setup

### CODE REVIEW

File: src/platform-includes/source-context/java.mdx [hunk 1]
```diff
@@ -194,6 +194,49 @@ sentry {
 
 </PlatformSection>
 
+
+<PlatformSection supported={["android"]}>
```
Comment (stefanosiano) [inline src/platform-includes/source-context/java.mdx:198]: @romtsn these bits are relevant for Android only, right?

File: src/platform-includes/source-context/java.mdx [hunk 2]
```diff
@@ -277,6 +332,18 @@ Next, upload that source bundle to Sentry:
...
 ```
 
+<PlatformSection supported={["android"]}>
+
+### Uploading including the Native Source Bundle
+
+Include the native source bundle with `--include-sources` and upload it to Sentry:
+
+```
+sentry-cli debug-files upload --include-sources /path/to/files --type jvm path/to/bundle
```
Comment (stefanosiano) [inline src/platform-includes/source-context/java.mdx:342]: hope i got the command right here

File: src/platform-includes/source-context/java.mdx [hunk 3]
```diff
@@ -277,6 +331,18 @@ Next, upload that source bundle to Sentry:
...
+```
+
+</PlatformSection>
```
Comment (markushi) [inline src/platform-includes/source-context/java.mdx:344]: I'm not an expert but I just tried to upload the native symbols + sources for our Android demo app. It works a bit easier for the NDK, there's no need to create a source bundle - just adding `--include-sources` does the trick.

How about the following? (GH suggestions is not working due to code blocks):

### Uploading Native Symbols and Source Code

Upload the native symbols manually using the `debug-files upload` command, here's a typical example for Android:

```
sentry-cli debug-files upload app/build/intermediates/merged_native_libs/
```

In order to upload the native source files as well, you can use the `--include-sources` flag. Using this flag sentry-cli will automatically generate source bundles on the fly during the upload.

```
sentry-cli debug-files upload app/build/intermediates/merged_native_libs/ --include-sources
```

File: src/platform-includes/source-context/java.mdx [hunk 4]
```diff
@@ -269,6 +311,18 @@ First, create the source bundle containing your source files:
 sentry-cli debug-files bundle-jvm --output path/to/store/bundle --debug-id A_VALID_UUID path/to/source-code
 ```
 
+<PlatformSection supported={["android"]}>
```
Comment (markushi) [inline src/platform-includes/source-context/java.mdx:314]: I think we could skip the section if we use the `--include-sources` flag I mentioned below.

File: src/platform-includes/source-context/java.mdx (hunk 2 above)
Comment (markushi) [reply src/platform-includes/source-context/java.mdx:341]: I don't think it works this way, at least the Android Gradle Plugin performs two uploads: One for JVM and one for Native sources / symbols - I've added a comment below how it works.

**Review by markushi — CHANGES_REQUESTED** [review]
Looking good, please have a look at my comments

File: src/platform-includes/source-context/java.mdx (hunk 4 above)
Comment (romtsn) [reply src/platform-includes/source-context/java.mdx:314]: right, we don't need this one

File: src/platform-includes/source-context/java.mdx (hunk 3 above)
Comment (romtsn) [reply src/platform-includes/source-context/java.mdx:344]: that's how it works yes 👍 sentry-cli will walk the file tree on itself and search for relevant sources when `--include-sources` is supplied.

**Comment by romtsn** [comment]
@stefanosiano  I still think the current structure might be confusing - there's a bunch of unrelated to NDK subsections in "Manual Upload". Should we just have two h1 headings `Source Context for Java/Kotlin` and `Source Context for NDK` so they are better separated?
image

**Comment by stefanosiano** [comment]
> @stefanosiano I still think the current structure might be confusing - there's a bunch of unrelated to NDK subsections in "Manual Upload". Should we just have two h1 headings `Source Context for Java/Kotlin` and `Source Context for NDK` so they are better separated? image
> 
> E.g. UUIDs, Know Limitations, Configuring the SDK - all of that is not needed for native sources

Now there are only 2 sections related to NDK, one for gradle plugin and one for manual upload.
I get everything else is not needed for the NDK, but i think someone uploading native source wants to upload java source first.
I mean NDK is more something to upload after the other things are configured.
I can easily make a separated section for the NDK and put it at the end, if you think it's less confusing. 
My only fear is that moving to a separate section means moving both manual and gradle plugin info (only the part related to ndk) to the bottom. And if i move the gradle plugin, either i have to duplicate the whole configuration (org, project, etc.) or the user would have to go up to fetch the full configuration and back down again to continue for the ndk

**Review by markushi — APPROVED** [review]
LGTM!

**Comment by romtsn** [comment]
@stefanosiano I mean this looks like you have to still configure the SDK after uploading NDK source - but this is not the case, the `Configuring the SDK` section is only relevant for manual upload of Java/Kotlin sources 
image

**Comment by stefanosiano** [comment]
@romtsn I moved the NDK manual upload to the end as a h2 section, separated from the previous one. Having a h1 for that seems too much imho

File: src/platform-includes/source-context/java.mdx [hunk 5]
```diff
@@ -310,3 +352,21 @@ sentry.bundle-ids=A_VALID_UUID`
 ```Java
 options.addBundleId("A_VALID_UUID");
 ```
+
+<PlatformSection supported={["android"]}>
+
+## Manually Uploading Native Symbols and Source Code
```
Comment (marandaneto) [inline src/platform-includes/source-context/java.mdx:358]:
```suggestion
## Manually Uploading Debug Symbols and Source Code
```

Comment (marandaneto) [reply src/platform-includes/source-context/java.mdx:358]: I'd also change every place you mention `native symbols` to `debug symbols` in this page.

File: src/platform-includes/source-context/java.mdx [hunk 6]
```diff
@@ -310,3 +352,21 @@ sentry.bundle-ids=A_VALID_UUID`
...
+
+Upload the native symbols manually using the `debug-files upload command`, here's a typical example for Android:
+
+```
+sentry-cli debug-files upload app/build/intermediates/merged_native_libs/
+```
```
Comment (marandaneto) [inline src/platform-includes/source-context/java.mdx:364]: I'd rather skip this to make it simpler and go directly with the example below.
You can adapt the wording if necessary.

Comment (stefanosiano) [reply src/platform-includes/source-context/java.mdx]: what about now?

**Pull request merged**